contracts/ethereum/l1Aggregator.sol

```diff
@@ -39,25 +41,32 @@ contract L1Aggregator {
     function sendData(address asset, address bridge) public {
         require(bridges[asset] != address(0), "no bridge found for the asset");
         // asset with zero address corresponds to wrapped ether on Starknet
+        bool isPublised = assetPublishedForPeriod[asset][block.number];
```

Comment:
you can do it in one line `if (assetPublishedForPeriod[asset][block.number] ) revert AlreadyPublished();`